core/ingest/src/main/java/mil/nga/giat/geowave/core/ingest/IngestUtils.java

```diff
@@ -62,23 +84,213 @@
 					LOGGER.warn("Adapter not found for " + geowaveData.getValue());
 					continue;
 				}
-				final IndexWriter indexWriter;
-				if (idx.getId().equals(
-						geowaveData.getIndexId())) {
-					indexWriter = primaryIndexWriter;
-				}
-				else {
-					final PrimaryIndex index = requiredIndexMap.get(geowaveData.getIndexId());
-					if (index == null) {
-						LOGGER.warn("Index '" + geowaveData.getIndexId().getString() + "' not found for " + geowaveData.getValue());
-						continue;
+				IndexWriter indexWriter;
+				for (final ByteArrayId indexId : geowaveData.getIndexIds()) {
+					indexWriter = dimensionalityIndexMap.get(indexId);
+					if (indexWriter == null) {
+						final PrimaryIndex index = requiredIndexMap.get(indexId);
+						if (index == null) {
+							LOGGER.warn("Index '" + indexId.getString() + "' not found for " + geowaveData.getValue());
+							continue;
+						}
+						indexWriter = ingestRunData.getIndexWriter(index);
 					}
-					indexWriter = ingestRunData.getIndexWriter(index);
+					indexWriter.write(
+							adapter,
+							geowaveData.getValue());
+				}
+			}
+		}
+	}
+
+	public static PrimaryIndex getIndex(
```

Comment:
I kinda want a tool that can be used to recommend an index configuration.  The minimum requirement for this would be a representative set of queries, in which case we could use SFC cardinality ratios much like the best fit query planner you worked to recommend an SFC config that constrains the query the most (using a cardinality estimate similar to the choosebestindex estimator).  Additionally, as a more precise option you could give a set of files that would hypothetically be ingested, but rather than ingest them, we could just run them through some form of histogram statistic that can give good estimates for counts that match the key range represented by a query constraint for indices to inform the choice of index...

really the first route seems pretty straightforward, the second with the included histogram is a bit more complex